platform/api/scaleway/api.go

```diff
@@ -0,0 +1,96 @@
+// Copyright The Mantle Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package scaleway
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/scaleway/scaleway-sdk-go/api/instance/v1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+
+	"github.com/flatcar/mantle/platform"
+	maws "github.com/flatcar/mantle/platform/api/aws"
+)
+
+var (
+	endpoint = "https://s3.%s.scw.cloud"
+)
+
+// Options hold the specific Scaleway options.
+type Options struct {
+	*platform.Options
+	// AccessKey is the Scaleway access key in the AWS format.
+	// Get the credentials at https://console.scaleway.com/iam/api-keys
+	AccessKey string
+	// Image is the ID of the Scaleway image to deploy.
+	Image string
+	// InstanceType is the type of the instance (e.g DEV1-S).
+	InstanceType string
+	// OrganizationID is the ID of the organization.
+	// Get the ID at https://console.scaleway.com/organization/settings
+	OrganizationID string
+	// ProjectID is used as "default" project for all requests.
+	ProjectID string
+	// Region is used as "default" region for all requests.
+	Region string
+	// SecretKey is the Scaleway secret key in the AWS format.
+	// Get the credentials at https://console.scaleway.com/iam/api-keys
+	SecretKey string
+	// Zone is used as "default" zone for all requests.
+	Zone string
+}
+
+// API is a wrapper around Scaleway instance API and
+// S3 AWS API (for object storage operation).
+type API struct {
+	opts *Options
+	*maws.API
```

Comment:
Yep, that's icky. Maybe we could consider (in future, obviously) splitting out S3 handling into a separate package, so AWS and Scaleway can use it.